src/Bedrock.Framework.Experimental/Protocols/Kafka/PayloadWriterContext.cs

```diff
@@ -0,0 +1,43 @@
+﻿#nullable enable
+#pragma warning disable CA1815 // Override equals and operator equals on value types
+
+using System;
+using System.Collections.Generic;
+using System.IO.Pipelines;
+using System.Runtime.CompilerServices;
+
+namespace Bedrock.Framework.Experimental.Protocols.Kafka
+{
+    public class PayloadWriterContext
```

Comment:
The context is intended to be passed around, and contains non-struct friendly fields.